### PR TITLE
Graduate apis to v1alpha2

### PR DESCRIPTION
**Pull request by ahg-g**
#### What type of PR is this?

/kind api-change

#### What this PR does / why we need it:

We changed the namespaced object API Queue to LocalQueue; while based on k8s API deprecation policy we don't necessarily need to bump the API version (since it is technically removing and adding an API); we elected to do it to emphasize the change.

#### Which issue(s) this PR fixes:

Part of #330

### CODE REVIEW

**Review by alculquicondor — COMMENTED** [review]
Please add a release note in the CHANGELOG folder

**Comment by ahg-g** [comment]
> Please add a release note in the CHANGELOG folder

done.

File: CHANGELOG/CHANGELOG-0.2.md [hunk 1]
```diff
@@ -4,3 +4,4 @@ Changes since `v0.1.0`:
 
 - Fixed bug in a BestEffortFIFO ClusterQueue where a workload might not be
   retried after a transient error.
+- Bumped the API version from v1alpha1 to v1alpha2. v1alpha1 is no longer supported.
```
[the remark this answers is not in the document]
Comment (ahg-g) [reply CHANGELOG/CHANGELOG-0.2.md]: done.

**Pull request merged**